Lib/test/test_os.py

```diff
@@ -230,6 +230,54 @@ def test_read(self):
             self.assertEqual(type(s), bytes)
             self.assertEqual(s, b"spam")
 
+    def test_readinto(self):
```

Comment:
Maybe add some tests for type errors as well? namel `test_readinto_exceptions` which checks that `os.readinto("abc")` for instance raises a TypeError or something like this (I don't know if we can have other exceptions than TypeErrors here)

Reply:
I think a `open()` "file object" passed in would definitely be good to have.  Trying to avoid testing "did clinic translate right" (that seems like clinic-specific argument conversion tests), but definitely "does the clinic input have the right type constraints" makes sense to me. The `Py_buffer(accept={rwbuffer})` I test a bit here, but definitely I'm lacking `fd` tests currently.

Reply:
`test_os.FileTests.test_readinto_badarg` covers some cases here.

Reply:
I would still duplicate them for `os.readinto`. That way, we're sure we're not introducing undesired regressions if we ever change the implementation in the future. Or test `os.readinto` in the tests out there if it's possible.

Is it actually possible to create a buffer with a negative length from pure Python (maybe with a fake iterable class or something like this)? if so, it's also good to exercise the check on the buffer length. If it requires the C API, just forget about the test.

Reply:
I can't find any tests that do currently. `test_buffer` does make some buffer objects using `_testcapi`. It feels like something that should be added to https://github.com/python/cpython/blob/main/Lib/test/test_buffer.py#L4619-L4638 potentially